### ISSUE TITLE
LocalDaskExecutor multiprocessing regression due to dask-core==2021.4.0

### ISSUE THREAD

**Issue by kpweiler** [PrefectHQ/prefect#4537]
## Description
When using `LocalDaskExecutor`, tasks that can be run in parallel are not run in parallel if the installed `dask-core` library is `2021.4.0` or later.

## Expected Behavior
Parallel tasks should be executed in parallel

## Reproduction
```
from prefect import task, Flow
from prefect.executors import LocalDaskExecutor
import time

@task
def run_thing1():
    time.sleep(5)
    return True

@task
def run_thing2():
    time.sleep(5)
    return True

if __name__ == "__main__":
    with Flow("test_flow", executor=LocalDaskExecutor(scheduler="processes")) as flow:
        run_thing1()
        run_thing2()

    flow.run()
```

When this is executed with `dask-core<=2021.3.1`, the output is as follows:
```
[2021-05-16 20:27:11-0500] INFO - prefect.FlowRunner | Beginning Flow run for 'test_flow'
[2021-05-16 20:27:11-0500] INFO - prefect.TaskRunner | Task 'run_thing1': Starting task run...
[2021-05-16 20:27:11-0500] INFO - prefect.TaskRunner | Task 'run_thing2': Starting task run...
[2021-05-16 20:27:16-0500] INFO - prefect.TaskRunner | Task 'run_thing1': Finished task run for task with final state: 'Success'
[2021-05-16 20:27:16-0500] INFO - prefect.TaskRunner | Task 'run_thing2': Finished task run for task with final state: 'Success'
[2021-05-16 20:27:16-0500] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
```

When this is run with `dask-core>=2021.4.0`, however, the output is as follows:
```
[2021-05-16 20:26:21-0500] INFO - prefect.FlowRunner | Beginning Flow run for 'test_flow'
[2021-05-16 20:26:22-0500] INFO - prefect.TaskRunner | Task 'run_thing1': Starting task run...
[2021-05-16 20:26:22-0500] INFO - prefect.TaskRunner | Task 'run_thing2': Starting task run...
[2021-05-16 20:26:27-0500] INFO - prefect.TaskRunner | Task 'run_thing1': Finished task run for task with final state: 'Success'
[2021-05-16 20:26:27-0500] INFO - prefect.TaskRunner | Task 'run_thing2': Finished task run for task with final state: 'Success'
[2021-05-16 20:26:27-0500] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
```

## Environment
```
{
  "config_overrides": {},
  "env_vars": [],
  "system_information": {
    "platform": "Linux-4.15.0-140-generic-x86_64-with-debian-buster-sid",
    "prefect_backend": "server",
    "prefect_version": "0.14.19",
    "python_version": "3.7.10"
  }
}
```

**Comment by mgsnuno** [comment]
can this be because of https://github.com/dask/dask/pull/6322 introduced in `dask==2021.04.0`?
also maybe the cause of https://github.com/PrefectHQ/prefect/issues/4555

**Comment by mgsnuno** [comment]
The reason why I bumped into this is because in the last month sometimes I find that I have flow running for 2 days already and is stuck in a mapped task that reads parquet files using dask and inserts them into a DB. I use DaskExecutor. Everything became worse after I updated to the latest dask/distributed 2021.5.0

**Comment by mgsnuno** [comment]
Hi. Anyone experiencing this issues?

**Comment by vholmer** [comment]
@mgsnuno I'm also having these issues unfortunately! I'm running prefect core 0.14.19 agents via prefect cloud using docker. I'm using the code supplied in prefect/executors/dask.py which is, if I understand the code correctly, not directly from the dask repository itself but something that the prefect team manages.

Here's an example (not very minimal but it's what I can show you) of my issues:

mve.py
```
from prefect import Flow, Parameter
from prefect.tasks.prefect import StartFlowRun
from prefect.executors import LocalDaskExecutor

from orchestration.flows.sleep import FLOW_NAME as SLEEP

FLOW_NAME = "MVE"

def create_flow(environment) -> Flow:
    project_name = "My Project"

    run_name = "MVE Run"

    sleep1 = StartFlowRun(
        flow_name=SLEEP, project_name=project_name, run_name=run_name, wait=True
    )

    sleep2 = StartFlowRun(
        flow_name=SLEEP, project_name=project_name, run_name=run_name, wait=True
    )

    with Flow(FLOW_NAME, schedule=None, executor=LocalDaskExecutor()) as flow:
        dummy_param = Parameter(name="Dummy", default="Dummy")

        sleep1.set_upstream(dummy_param)
        sleep2.set_upstream(dummy_param)

    return flow

def create_env(environment: str, env_params: dict, common_params: dict):
    """
    Create labels and environment variables needed to run flow
    """
    labels = [environment, "transformation"]
    docker_conf = {
        "docker_env_vars": {},
    }
    return labels, docker_conf
```

sleep.py
```
from prefect import Flow
from prefect import task

import time

FLOW_NAME = "Sleep"

@task
def task_sleep():
    time.sleep(20)

def create_flow(environment) -> Flow:
    with Flow(FLOW_NAME, schedule=None) as flow:
        task_sleep()

    return flow

def create_env(environment: str, env_params: dict, common_params: dict):
    """
    Create labels and environment variables needed to run flow
    """
    labels = [environment, "transformation"]
    docker_conf = {
        "docker_env_vars": {},
    }
    return labels, docker_conf
```
And here's the execution in Prefect cloud which is not parallelized at all:

![image](https://user-images.githubusercontent.com/5623158/120340115-02d21d80-c2f6-11eb-8457-bd054e2a8ffe.png)

Let me know if you need further information here!

**Comment by madkinsz** [comment]
Hey all,

Can we clarify if this is something that is limited to the `LocalDaskExecutor` or if there are also concurrency issues on the `DaskExecutor` ?

**Comment by vholmer** [comment]
@madkinsz It's both! I tried the non-local executor with the same result.

**Comment by madkinsz** [comment]
I can reproduce this with

- `LocalDaskExecutor(scheduler="processes")`
- `LocalDaskExecutor(scheduler="processes", num_workers=4)`

I cannot reproduce this with

- `LocalDaskExecutor(scheduler="threads")`
- `DaskExecutor()`

**Comment by nikolabold** [comment]
I have experienced the same issue - `LocalDaskExecutor` with the `scheduler="processes"` parameter runs mapped tasks sequentially. When running with `scheduler="processes"` no such issues occur.

After downgrading dask to version `2021.3.0` the problem was resolved.

**Comment by madkinsz** [comment]
This is indeed introduced by #6322 which adds a default `chunksize` of 6 which means a trivial flow like this will not submit tasks to run in parallel. This is to offset the cost of spinning up additional processes. Since we're managing the pool ourselves, I may set the chunksize to 1 for now but I'll have to look into the implications further. To be clear, you're only losing parallelism in trivial flows like this and a larger flow will still execute in parallel as larger chunks are submitted.

See related dask issue at https://github.com/dask/dask/issues/7510

**Comment by mgsnuno** [comment]
@madkinsz my issue is really that mapped tasks sometimes just hang. In the end I'm inclined that it's not a prefect issue, something is dask.distributed that gets dead locked.

**Comment by madkinsz** [comment]
@mgsnuno -- to discuss that, you should open a separate issue (perhaps on the dask-distributed github) since this is unrelated to the thread you are commenting in. I would recommend looking at the dask scheduler UI.

**Comment by jakirkham** [comment]
Just to add here, one can still create a `multiprocessing.Pool` and use that with Dask if preferred.

```python
import multiprocessing

import dask
import dask.array


with dask.config.set(pool=multiprocessing.Pool(4)):
    a = dask.array.arange(1_000_000, chunks=10_000)
    r = a.sum()
    print(r.compute())
```
